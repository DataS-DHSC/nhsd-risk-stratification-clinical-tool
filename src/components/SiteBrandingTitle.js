import PropTypes from 'prop-types';
import {
  SITE_BRANDING_TITLE,
  MHRA_REGISTERED_NAME,
} from '../constants/site-branding';

const SiteBrandingTitle = ({ isMHRARegisteredName, isDefaultName }) => (
  <>
    {isMHRARegisteredName && !isDefaultName
      ? MHRA_REGISTERED_NAME
      : SITE_BRANDING_TITLE}
  </>
);

SiteBrandingTitle.propTypes = {
  isMHRARegisteredName: PropTypes.bool,
  isDefaultName: PropTypes.bool,
};

SiteBrandingTitle.defaultProps = {
  isMHRARegisteredName: false,
  isDefaultName: false,
};

export default SiteBrandingTitle;
