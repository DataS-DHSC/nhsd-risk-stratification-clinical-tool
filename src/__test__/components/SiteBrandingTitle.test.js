import React from 'react';
import { render } from '@testing-library/react';
import SiteBrandingTitle from '../../components/SiteBrandingTitle';

const setup = (isMHRARegisteredName) => {
  const props = {
    isMHRARegisteredName,
  };

  return render(<SiteBrandingTitle {...props} />);
};

describe('SiteBrandingTitle', () => {
  it('renders correctly for site branding title', () => {
    const container = setup(false);
    expect(container.asFragment()).toMatchSnapshot();
  });

  it('renders correctly for MHRA name', () => {
    const container = setup(true);
    expect(container.asFragment()).toMatchSnapshot();
  });
});
