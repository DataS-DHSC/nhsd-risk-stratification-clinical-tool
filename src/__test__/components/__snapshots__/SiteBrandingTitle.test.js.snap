// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteBrandingTitle renders correctly for MHRA name 1`] = `
<DocumentFragment>
  COVID-19 Risk Assessment Service
</DocumentFragment>
`;

exports[`SiteBrandingTitle renders correctly for site branding title 1`] = `
<DocumentFragment>
  COVID-19 Clinical Risk Assessment Tool
</DocumentFragment>
`;
